### ISSUE TITLE
question about pop and generate example

### ISSUE THREAD

**Issue by lullcant** [potamides/DaTikZ#1]
The first issue is that when I run the main.py. It shows:
 sketchify = args.pop("sketchify")
AttributeError: 'Namespace' object has no attribute 'pop'

The second issue is that why the script continue to generate a lot of training split examples (thousands of example) even I only have one 
jsonl file ? I guess there are only around 70 Tikz figure in the jsonl. I run the main.py with the shell script.
```
DATIKZ_ARXIV_FILES=(/mnt/d/arxiv/extracted/arXiv_src_2310_001.jsonl) 
python main.py --arxiv_files "${DATIKZ_ARXIV_FILES[@]}" --size 334 
```
The script continue to show Generating train split: xxxx examples and it never ends. What is the example means here.

Thanks in advance

**Comment by lullcant** [comment]
Also I’d like to ask if I only want the tikz code of the arxiv jsonl file list I extracted, how can I get it.
Thanks

**Comment by potamides** [comment]
>The first issue is that when I run the main.py. It shows:
>sketchify = args.pop("sketchify")
>AttributeError: 'Namespace' object has no attribute 'pop'

Thanks for the report! I will push a fix soon.

>Also I’d like to ask if I only want the tikz code of the arxiv jsonl file list I extracted, how can I get it.

By default, the script extracts tikzpictures from a lot of sources apart from arxiv. You can disable that by commenting out the corresponding lines here:

https://github.com/potamides/DaTikZ/blob/e9c5f5e3dac5c85e3d5e1b6682705293c6283b20/datikz/datikz.py#L140-L150

And optionally also here:

https://github.com/potamides/DaTikZ/blob/e9c5f5e3dac5c85e3d5e1b6682705293c6283b20/datikz/datikz.py#L131-L137

**Comment by lullcant** [comment]
There is another issue that when
```
  File "/home/DaTikZ/main.py", line 108, in <module>
    datikz = load_dataset("datikz", split="train", **vars(args))
  File "/usr/local/lib/python3.10/dist-packages/datasets/load.py", line 2108, in load_dataset
    ds = builder_instance.as_dataset(split=split, verification_mode=verification_mode, in_memory=keep_in_memory)
  File "/usr/local/lib/python3.10/dist-packages/datasets/builder.py", line 1125, in as_dataset
    datasets = map_nested(
  File "/usr/local/lib/python3.10/dist-packages/datasets/utils/py_utils.py", line 484, in map_nested
    mapped = function(data_struct)
  File "/usr/local/lib/python3.10/dist-packages/datasets/builder.py", line 1155, in _build_single_dataset
    ds = self._as_dataset(
  File "/usr/local/lib/python3.10/dist-packages/datasets/builder.py", line 1229, in _as_dataset
    dataset_kwargs = ArrowReader(cache_dir, self.info).read(
  File "/usr/local/lib/python3.10/dist-packages/datasets/arrow_reader.py", line 251, in read
    raise ValueError(msg)
ValueError: Instruction "train" corresponds to no data!
```
when I use the following shell script to run the code
```
DATIKZ_ARXIV_FILES=(/usr/arxiv/extracted/arXiv_src_2310_001.jsonl)
python main.py --arxiv_files "${DATIKZ_ARXIV_FILES[@]}" --size 334 
```

**Comment by lullcant** [comment]
There is another issue that
For some files, it will get stucked in the multiprocessing pool.
Can I set timeout to skip some problematic files? Thanks

**Comment by potamides** [comment]
Yeah, I experienced that issue as well. Strangely I think it only happens with Python 3.11+. Python 3.10 used to throw a recursion error, if I remember correctly. I am planning to release a larger cleanup of the code soon, in the meantime can you try it again with Python 3.10?

**Comment by lullcant** [comment]
actually, my python version is python 3.10.12. But it still get stucked

**Comment by potamides** [comment]
I just pushed a major refactor with a lot of enhancements and fixes, including for the problem you describe. Could you try again if it works now?
